app/livechat/client/views/app/tabbar/customerChatHistory.js

```diff
@@ -0,0 +1,230 @@
+		// will return pervious chats list
+		let history = Template.instance().history.get();
+		let newHisTory = []
+		for(i=1; i<history.length; i++){
+			var d = Date.parse(history[i].ts);
+			var D = new Date();
+			D.setTime(d);
+			var newTs = D.toTimeString();
+			var time = getTime(newTs);
+			history[i].time = time;
+			history[i].count = history[i].msgs-3;
+			newHisTory.push(history[i]);
+		}
+		Template.instance().modifiedHistory.set(newHisTory);
+		return Template.instance().modifiedHistory.get();
+	},
+	len(){
+		// will return length of total messages in room
+		len = Template.instance().historyResu.get();
+		return len = len.length-2;
+	},
+	title() {
+		
+		let title = moment(this.ts).format('L LTS');
+
+		if (this.label) {
+			title += ` - ${ this.label }`;
+		}
+		return title;
+	},
+});
+
+Template.customerChatHistory.onCreated(function() {
+	const currentData = Template.currentData();
+	this.visitorId = new ReactiveVar();
+	this.isLoading = new ReactiveVar(false);
+	this.modifiedHistory = new ReactiveVar([])
+	this.history = new ReactiveVar([]);
+	this.offset = new ReactiveVar(0);
+	this.total = new ReactiveVar(0);
+	this.isAllChat = new ReactiveVar(true);
+	this.isChatClicked = new ReactiveVar(true);
+	this.autorun(async () => {
+		const { room } = await APIClient.v1.get(`rooms.info?roomId=${ currentData.rid }`);
+		if (room && room.v) {
+			this.visitorId.set(room.v._id);
+		}
+	});
+
+	this.autorun(async () => {
+		
+		if (!this.visitorId.get() || !currentData || !currentData.rid) {
+			return;
+		}
+
+		const offset = this.offset.get();
+	
+		this.isLoading.set(true);
+		const { history, total } = await APIClient.v1.get(`livechat/visitors.chatHistory/room/${ currentData.rid }/visitor/${ this.visitorId.get() }?count=${ ITEMS_COUNT }&offset=${ offset }`);
+		
+		this.isLoading.set(false);
+		this.total.set(total);
+		this.history.set(this.history.get().concat(history));
+		// this will load all the chat history chat messages
+		allHistoryChat = [];
+		var limit = 100;
+		let count = this.history.get();
+		if(allHistoryChat.length>0){
+			return
+		}
+		for(i=0;i<count.length;i++){
+			let id = count[i]._id;
+			let token = count[i].v.token;
+			setTimeout(async function(){
+			const historyResult  =  await APIClient.v1.get(`livechat/messages.history/${ id }?token=${token}&limit=${limit}`);
+			let count2 = historyResult.messages;
+			
+			for(j=0; j<count2.length; j++){
+
+					allHistoryChat.push(count2[j]);
+				
+			}
+			
+		},1000)
+			
+		}
+		
+	
+	});	
+});
+
+Template.customerChatHistory.events({
+	'scroll .visitor-scroll': _.throttle(function(e, instance) {
+		if (e.target.scrollTop >= (e.target.scrollHeight - e.target.clientHeight)) {
+			const history = instance.history.get();
+			if (instance.total.get() <= history.length) {
+				return;
+			}
+			return instance.offset.set(instance.offset.get() + ITEMS_COUNT);
+		}
+	}, 200),
+	'keyup #searchInput': async function(event,template){ 
+		
+		searchResults = [];
```

Comment:
All of the code below is not necessary.

Reply:
have done some changes please check

Reply:
As I mentioned, all of this code below is unnecessary.
You need to create a `loadRooms` method that supports an optional parameter: `searchTerm`.
The method will call the `livechat/visitors.chatHistory/` endpoint. When you are searching, you will call `loadRooms(searchTerm)`.

Is not much more simple?
I shared with you [an example](https://github.com/RocketChat/Rocket.Chat/blob/develop/app/livechat/client/views/app/livechatDepartments.js#L81) on how to implement this, why not just follow this instruction?

There is no reason for this loop, well you call the endpoint, searching or not, then you'll get the `rooms` you need to render.
If you need to change the object before rendering, you can create a helper method to do it, the `this` will be the object you pass to the `chatRoomHistoryItem` template.